src/qibosoq/components.py

```diff
@@ -80,14 +81,33 @@ class Parameter(IntEnum):
     BIAS = auto()
     DURATION = auto()
 
+    @overload
+    @classmethod
+    def variants(cls, parameters: str) -> "Parameter":
+        """Convert a string to a Parameter."""
+
+    @overload
+    @classmethod
+    def variants(cls, parameters: Iterable[str]) -> Iterable["Parameter"]:
+        """Convert a iterable of str to an iterable of Parameters."""
+
+    @classmethod
+    def variants(cls, parameters):
+        """Convert from strings to Parameters."""
+        if isinstance(parameters, str):
+            return cls[parameters.upper()]
+        if isinstance(parameters, Iterable):
+            return type(parameters)(cls[par.upper()] for par in parameters)
+        raise NotImplementedError("Conversion not supported")
```

Comment:
This is not the most Pythonic way to check if something is iterable: something is iterable if you can iterate it.

So, if you don't care about the specific error, you could just write:
```suggestion
        return type(parameters)(cls[par.upper()] for par in parameters)
```
otherwise, catch the error with `try ... except`, and raise your favorite error.

Reply:
I'm not sure the constructor would work for any iterable. But if it does, why not :)

And even if it works for the types you're interested in, it's already good enough, and just better even if it works with something else.
This is in the spirit above: you don't check nor ask permission, you do it. And if some operation does not apply, it will raise its own error (this is the duck typing application, and the goal is to make your code cleaner, less checks and errors raised, but more general at the same time).